### PR TITLE
add output format arg for `cli.dags.trigger`

### LINKED ISSUE #29178: Add `output` format to missing cli commands
### Description

I have noticed that for some commands, there is an option to get the output in json or yaml (as described in this PR from 2020 https://github.com/apache/airflow/issues/12699).

However, there are still some commands that do not support the `--output` argument, most notable one is the `dags trigger`. 

When triggering a dag, it is crucial to get the run_id that has been triggered, so the triggered dag run can be monitored by the calling party. However, the output from this command is hard to parse without resorting to (gasp!) regex:
```
[2023-01-26 11:03:41,038] {{__init__.py:42}} INFO - Loaded API auth backend: airflow.api.auth.backend.session
Created <DagRun sample_dag @ 2023-01-26T11:03:41+00:00: manual__2023-01-26T11:03:41+00:00, state:queued, queued_at: 2023-01-26 11:03:41.412394+00:00. externally triggered: True>
```

As you can see, extracting the run_id `manual__2023-01-26T11:03:41+00:00` is not easy from the above output.

For what I see [in the code](https://github.com/apache/airflow/blob/main/airflow/cli/cli_parser.py#L1156), the `ARG_OUTPUT` is not added to `dag_trigger` command.

### Use case/motivation

At my company we want to be able to trigger dags from another airflow environment (mwaa) and be able to wait for its completion before proceeding with the calling DAG.

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by hussein-awala**
closes: #29178 

---

Add output format argument for the command `dags.trigger` to simplify parsing the created `DagRun` info.

### CODE REVIEW

File: airflow/api/client/local_client.py [hunk 1]
```diff
@@ -22,21 +22,39 @@
...
+    ) -> dict | None:
         dag_run = trigger_dag.trigger_dag(
             dag_id=dag_id,
             run_id=run_id,
             conf=conf,
             execution_date=execution_date,
             replace_microseconds=replace_microseconds,
         )
-        return f"Created {dag_run}"
+        if isinstance(dag_run, DagRun):
```
Comment (ephraimbuddy) [inline airflow/api/client/local_client.py]:
```suggestion
        if dag_run:
```
Do we need the isinstance?
Also let's add a test for this change in tests/api/client/test_local_client.py

**Review by ephraimbuddy — COMMENTED** [review]
Please add tests for the changes

File: tests/api/client/test_local_client.py [hunk 2]
```diff
@@ -128,6 +128,45 @@ def test_trigger_dag(self, mock):
...
+                "state": DagRunState.QUEUED,
+            }
+            dag_run = self.client.trigger_dag(dag_id=test_dag_id)
+            assert expected_dag_run == dag_run
+            mock.reset_mock()
+
+            # test output when no DagRun is created
+            mock.return_value = None
+            dag_run = self.client.trigger_dag(dag_id=test_dag_id)
+            assert None is dag_run
```
Comment (ephraimbuddy) [inline tests/api/client/test_local_client.py]:
```suggestion
            assert not dag_run
```

**Review by ephraimbuddy — APPROVED** [review]
cc @jedcunningham

**Review by eladkal — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

File: airflow/api/client/local_client.py [hunk 3]
```diff
@@ -28,15 +28,32 @@
...
             execution_date=execution_date,
             replace_microseconds=replace_microseconds,
         )
-        return f"Created {dag_run}"
+        if dag_run:
+            return {
+                "conf": dag_run.conf,
+                "dag_id": dag_run.dag_id,
+                "dag_run_id": dag_run.run_id,
+                "data_interval_end": dag_run.data_interval_start,
```
Comment (yupbank) [inline airflow/api/client/local_client.py:46]: what's the rational behind the swapping of  `data_interval_end -> dag_run.data_interval_start` and `data_interval_start -> dag_run.data_interval_end` here ?

Comment (hussein-awala) [reply airflow/api/client/local_client.py:46]: Clearly this is a bug, here is the fix: #32548
